### PR TITLE
Security checklist draft

### PR DESCRIPTION
**Pull request by pnavalnyy**
First version of Security Checklist document

### CODE REVIEW

**Review by falsanu — COMMENTED** [review]
I am unsure if we should add examples for our common tools eg. `npm audit`, the renovate bot, PHP-tools (composer-stuff), .net?

File: general/security-checklist.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Security checklist  
+### How to
+It's recommended to go through the checklist once per two weeks. Feel free to choose the proper period based on your project scale and life span.
+
+### Checklist
+- Currently used software versions contain no known vulnerabilities
+- All the modules and dependencies are up to date
+- Automated security and dependency checkers are in use
+- No credentials, secrets or other sensitive data exposed in the repo (use git leaks or similar)
```
Comment (GrzegorzMachnik) [inline general/security-checklist.md]: in the repo, other widely used placed like not secret variables in pipelines or ENV vars (use git leaks or similar)

File: general/security-checklist.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- All necessary credentials are in Passbolt(?) and are up to date
+- If regular pen-test/security check is necessary, make sure it's done
+- No resources are using HTTP instead of HTTPS
+- No eval() is used in the code
+
```
Comment (GrzegorzMachnik) [inline general/security-checklist.md:13]: - New functionality written from the previous check is the risk from a security perspective (i.e. new public endpoints with the possibility to post data by the user to solution) if yes ask for Pen test or check this place in the team.

File: general/security-checklist.md (hunk 1 above)
Comment (GrzegorzMachnik) [reply general/security-checklist.md]: or Wiki

File: general/security-checklist.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Security checklist  
+### How to
+It's recommended to go through the checklist once per two weeks. Feel free to choose the proper period based on your project scale and life span.
+
+### Checklist
+- Currently used software versions contain no known vulnerabilities
+- All the modules and dependencies are up to date
+- Automated security and dependency checkers are in use
+- No credentials, secrets or other sensitive data exposed in the repo (use git leaks or similar)
+- All necessary credentials are in Passbolt(?) and are up to date
```
Comment (frederik-b) [inline general/security-checklist.md]: I don't think we're using passbolt
also using git-crypt or git-secrets is totally fine for secrets imho

File: general/security-checklist.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+- If regular pen-test/security check is necessary, make sure it's done
+- No resources are using HTTP instead of HTTPS
```
Comment (frederik-b) [inline general/security-checklist.md]: kubernetes is made to use HTTP all over the place this should be limited to external usage

File: general/security-checklist.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+- No eval() is used in the code
```
Comment (frederik-b) [inline general/security-checklist.md]: while this is true,
this is very specific, language specific and specific to a command
when we'll add this we should probably add a couple of dozen others too
and in the end it'ld be probably best to gather all of the specific rules somewhere else and refence them here

File: general/security-checklist.md (hunk 1 above)
Comment (pnavalnyy) [reply general/security-checklist.md]: Added in.

File: general/security-checklist.md (hunk 2 above)
Comment (pnavalnyy) [reply general/security-checklist.md:13]: Added a shorter version as a new item.

File: general/security-checklist.md (hunk 3 above)
Comment (pnavalnyy) [reply general/security-checklist.md]: Removed Passbolt mention, added git tools to the list.

File: general/security-checklist.md (hunk 4 above)
Comment (pnavalnyy) [reply general/security-checklist.md]: Rephrased it to sound less restrictive.

File: general/security-checklist.md (hunk 5 above)
Comment (pnavalnyy) [reply general/security-checklist.md]: Removed it.

**Review by GrzegorzMachnik — APPROVED** [review]

**Pull request merged**